Proposal: UIDataTestMethod / Support for DataRows in UITestMethods

The DataTestMethod attribute is really helpful and makes writing tests a lot easier. However when developing tests that need to run on the UI thread, we have to use the UITestMethod attribute. This however leaves us with having to implement data driven tests ourselfes. Having a UIDataTestMethod attribute would be great to have to fill the gap between UI tests and non UI tests.

[AB#1408166](https://devdiv.visualstudio.com/0bdbc590-a062-4c3f-b0f6-9383f67865ee/_workitems/edit/1408166)

Was trying to test this inside the project by adding a UWP test app and have the use the libraries from the repository, however I can not build the project as it fails with the following error message:

```
1>CSC : error CS7032: Key file 'D:\Projects\testfx\scripts\build\key.snk' is missing the private key needed for signing
```

Maybe you guys now more about this or know a way to work around this?

--- a/src/TestFramework/TestFramework.Extensions/Attributes/WinUI_UITestMethodAttribute.cs
+++ b/src/TestFramework/TestFramework.Extensions/Attributes/WinUI_UITestMethodAttribute.cs
@@ -84,7 +84,7 @@ public class UITestMethodAttribute : TestMethodAttribute
                 {
                     try
                     {
-                        result = testMethod.Invoke([]);
+                        result = testMethod.Invoke(null);
                         taskCompletionSource.SetResult(null);
                     }
                     catch (Exception e)

